### PR TITLE
add ability to reset passwords

### LINKED ISSUE #28: Support password resets via email
Unless I'm missing something it doesn't look like password resets are supported yet. 

If you'd like I'd be happy to submit a PR to support it via the same 'generates_token_for' API that I currently have an PR for email confirmations for.

### PR DESCRIPTION
**Pull request by kylekeesling**
Here's my attempt at implementing password resets using the [Rails 7.1 `generates_token_for` API](https://github.com/rails/rails/pull/44189).

Closes #28.

### CODE REVIEW

**Comment by kylekeesling** [comment]
I also stubbed in some tests for both the Mailer and Controller but it's not going to be happy until we figure out a way to backport the feature.

Was also wondering, is it worth creating a notification email to tell the user that their password was changed? If so it shouldn't be too much more work to roll that in.

**Comment by excid3** [comment]
So I added the `backports.rb` file to hopefully contain backports in a single place. This feature is a module already, so we could copy that and have backports inject it.

**Comment by kylekeesling** [comment]
Ok, so I was able to get a backport working, but I could not figure out how to add it to `backports.rb`. 

Loading it requires an initializer hook when ActiveRecord is loading, and based on other code I was able to grok from other projects this was the way I was able to get it to work.

If you know of a way to contain this within the backport just let me know and I can make the adjustment.

**Comment by kylekeesling** [comment]
(as I also said on #30) It's been a minute since I've looked at this, but I'm happy to update it now that 7.1 is on the horizon if you'd still find it useful!

**Comment by excid3** [comment]
@kylekeesling sure! I was going to but I need to focus on Rails World talk first. 🙃

I decided to require 7.1+ so we no longer need to mess with backporting which was getting messy.

**Comment by kylekeesling** [comment]
@excid3 I just pulled from `main` to refresh my PRs (#29 and #30) and fixed all the failing tests, so I think we are in a pretty good spot.

Good luck getting the talk together, and safe travels. I wish I could make it this year, I'm sure it'll be a lot of fun!

**Comment by excid3** [comment]
That was fast! Will check those out shortly.

And thanks! I am getting anxious with how fast Rails World is approaching. 😅

File: lib/revise_auth/routes.rb [hunk 1]
```diff
@@ -12,6 +12,8 @@ def revise_auth
         patch "profile/email", to: "email#update"
         patch "profile/password", to: "password#update"
 
+        resources :password_resets, param: :token, only: [:new, :create, :edit, :update]
```
Comment (excid3) [inline lib/revise_auth/routes.rb:15]: I always forget about `param:`!

Comment (kylekeesling) [reply lib/revise_auth/routes.rb:15]: Yeah, it's a pretty handy way to keep things tidy/explicit :)

**Pull request merged**